### PR TITLE
Closes #115 confirmation submission date

### LINKED ISSUE #115: Add timestamp for confirmation
Confirmations of multisig transactions should have a timestamp indicating when the confirmation was submitted.

### PR DESCRIPTION
**Pull request by jpalvarezl**
Closes #115 

- Added optionally present `submission_date`

### CODE REVIEW

File: src/models/converters/transactions/tests/details.rs [hunk 1]
```diff
@@ -70,10 +70,12 @@ fn multisig_custom_transaction_to_transaction_details() {
                     MultisigConfirmation {
                         signer: String::from("0x65F8236309e5A99Ff0d129d04E486EBCE20DC7B0"),
                         signature: Some(String::from("0x83b1506c409918f21031e93ed2f62310a5e0c05b1be89242a6a266a7de4af7bc6094e206b33387b8d4465af6087a4d2158815e613aeb186d88d9a1973e00bbe81b")),
+                        submission_time: multisig_tx.confirmations.as_ref().and_then(|confirmation| confirmation.get(0)?.submission_date.map(|it| it.timestamp_millis())),
```
Comment (rmeissner) [inline src/models/converters/transactions/tests/details.rs:73]: I would have hardcoded the number :D

File: src/models/service/transactions/details.rs [hunk 2]
```diff
@@ -37,6 +37,7 @@ pub struct MultisigExecutionDetails {
 pub struct MultisigConfirmation {
     pub signer: String,
     pub signature: Option<String>,
+    pub submission_time: Option<i64>,
```
Comment (rmeissner) [inline src/models/service/transactions/details.rs:40]: this should not be optional

**Review by rmeissner — CHANGES_REQUESTED** [review]

File: src/models/converters/transactions/tests/details.rs (hunk 1 above)
Comment (jpalvarezl) [reply src/models/converters/transactions/tests/details.rs:73]: I don't know how to do that in tests :/ parsing dates with `chrono` is a bit of a pain. I would like to clean that up eventually.

File: src/models/service/transactions/details.rs (hunk 2 above)
Comment (rmeissner) [reply src/models/service/transactions/details.rs:40]: so I would expect this https://github.com/gnosis/safe-client-gateway/blob/ca68312297009dfe934043de0d96e674a41d3779/src/models/backend/transactions.rs#L86 also to be not optional

Comment (jpalvarezl) [reply src/models/service/transactions/details.rs:40]: It is optional from the backend: https://github.com/gnosis/safe-client-gateway/blob/8491e453ea2e367300bd4098f87a5025e14d4060/src/models/backend/transactions.rs#L86

Comment (jpalvarezl) [reply src/models/service/transactions/details.rs:40]: `confirmations` is a string in the `Multisig` variant in the docs 🤔 

https://safe-transaction.rinkeby.gnosis.io/

Comment (rmeissner) [reply src/models/service/transactions/details.rs:40]: submission_date will always be present. See https://github.com/gnosis/safe-transaction-service/blob/master/safe_transaction_service/history/serializers.py#L267 (`created` is a database field that is set upon db object creation and will always be present)

File: src/models/converters/transactions/tests/details.rs (hunk 1 above)
Comment (rmeissner) [reply src/models/converters/transactions/tests/details.rs:73]: what do you mean? you know the expected number by converting the string `2020-06-22T14:58:34.055882Z` with some tool (e.g. https://codechi.com/dev-tools/date-to-millisecond-calculators/) ... and then you get: `1592837914055`

File: src/models/service/transactions/details.rs (hunk 2 above)
Comment (jpalvarezl) [reply src/models/service/transactions/details.rs:40]: alright, will remove the `Option<>` wrapper.

File: src/models/converters/transactions/tests/details.rs (hunk 1 above)
Comment (jpalvarezl) [reply src/models/converters/transactions/tests/details.rs:73]: aaah, I was trying to find how the deserialiser turns them into `DateTime<Utc>` , that's much better, thanks!

**Review by rmeissner — APPROVED** [review]

**Pull request merged**